lib/core/device_drivers.rb

```diff
@@ -191,4 +199,31 @@ def build_chrome_driver(chrome_ops)
     end
     return driver
   end
+
+  # merge case and/or config capabilities/options for Firefox
+  def merge_firefox_ops(config_caps, case_caps)
+    return merge_ops("firefoxOptions", config_caps, case_caps)
+  end
+
+  # build the Firefox driver, given a set of options
+  def build_firefox_driver(firefox_ops)
+    if @url.nil?
+      localFirefoxOptions = Selenium::WebDriver::Firefox::Options.new(
+        options: firefox_ops,
+      )
+      driver = Selenium::WebDriver.for(
+        :firefox, options: localFirefoxOptions
+      )
+    else
+      # remote selenium grid
+      log_debug("Selenium Server URL: #{@url}")
+      remoteFirefoxOptions = Selenium::WebDriver::Remote::Capabilities.firefox(
+        firefox_ops,
```

Comment:
this may not work, the chrome syntax uses `goog:chromeOptions`. But I suppose it's not too relevant at the moment